Source/Core/DiscIO/VolumeDirectory.cpp

```diff
@@ -497,4 +505,20 @@ static u32 ComputeNameSize(const File::FSTEntry& parent_entry)
   return name_size;
 }
 
+static char ASCIILowercaseChar(char c)
+{
+  if (c >= 'A' && c <= 'Z')
+    c += 0x20;
+  return c;
+}
+
+static std::string ASCIILowercaseString(const std::string& str)
+{
+  std::string copy = str;
+  // We can't use tolower, because its behavior depends on the locale,
+  // and we want to be deterministic
+  std::transform(copy.begin(), copy.end(), copy.begin(), ASCIILowercaseChar);
```

Comment:
You can use a lambda or `std::bind`.

You can technically use `std::bind` to pass functions with more parameters than what a predicate would pass through, if you know what data the other parameters need beforehand. You'd then mark the parameter the predicate is supposed to fill with a placeholder. It would look like so:

```cpp

std::string ASCIIToLowerString(std::string str)
{
  std::transform(str.begin(), str.end(), str.begin(),
                 std::bind(std::tolower<std::string::value_type>, std::placeholders::_1, std::locale::classic()));
  return str;
}
```

std::tolower looks ugly here, because two std::tolower overloads are present and you need to specify which one somehow. In this case, I'd just use a lambda, personally.